src/utils/hashmap.c

```diff
@@ -51,33 +51,32 @@ void hashmap_create_cb(hashmap *hm, hashmap_free_cb free_cb) {
  * All existing key-value pairs are rehashed, so this has some CPU impact.
  */
 static void hashmap_resize(hashmap *hm, unsigned int new_size) {
-    // Do not resize if equal size was requested
-    if(new_size == hm->capacity || new_size < 1)
+    if(new_size <= hm->capacity)
         return;
 
     // Allocate and zero out a new memory blocks for the resized bucket list
-    hashmap_node **new_buckets = omf_calloc(new_size, sizeof(hashmap_node *));
+    size_t old_bytes = hm->capacity * sizeof(hashmap_node *);
+    size_t new_bytes = new_size * sizeof(hashmap_node *);
+    hm->buckets = omf_realloc(hm->buckets, new_bytes);
+    memset(((void *)hm->buckets) + old_bytes, 0, new_bytes - old_bytes);
```

Comment:
Hmmh. I think pointer arithmetic on void pointer is undefined behaviour. Is the buckets pointer void? Should change it to unsigned char.

Reply:
Okay, checked that it has the type `hashmap_node **`, so could something like `memset(&hm->buckets[hm->capacity], 0, new_bytes - old_bytes);` work here?

Reply:
Checked that pointer arithmetic on void * is supported by gcc, so maybe don't need to change it.